src/components/ui-shell/navigation-menu-panel.js

```diff
@@ -0,0 +1,91 @@
+import mixin from '../../globals/js/misc/mixin';
+import createComponent from '../../globals/js/mixins/create-component';
+import initComponentByLauncher from '../../globals/js/mixins/init-component-by-launcher';
+import eventedShowHideState from '../../globals/js/mixins/evented-show-hide-state';
+import handles from '../../globals/js/mixins/handles';
+import eventedState from '../../globals/js/mixins/evented-state';
+import toggleAttribute from '../../globals/js/misc/toggle-attribute';
+import settings from '../../globals/js/settings';
+
+export default class NavigationMenuPanel extends mixin(
+  createComponent,
+  initComponentByLauncher,
+  eventedShowHideState,
+  handles,
+  eventedState
+) {
+  createdByLauncher = event => {
+    const isExpanded = !this.element.hasAttribute('hidden');
+    const newState = isExpanded ? 'collapsed' : 'expanded';
+    this.triggerButton = event.delegateTarget;
+    this.changeState(newState);
+  };
+
+  /**
+   *
```

Comment:
Nitpick - Do we want to update the comment here...?

Reply:
updated!